### PR TITLE
Fix the API docs of install_jax

### LINKED ISSUE #2728: Update the docstring of `install_jax`
Also, I see that the docstring is a bit misleading. The docstring uses `>>>`, misleading users into executing it in the `python` interpreter (a PR fixing it would be very welcome!) -

https://github.com/pybamm-team/PyBaMM/blob/4f03df423bc63f9b4bd5f7e926c78a2ad1af1f87/pybamm/util.py#L330-L341

Also, https://pybamm.readthedocs.io/en/latest/source/api/util.html misses the API documentation for install_jax. The function can be added here: https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/api/util.rst.

_Originally posted by @Saransh-cpp in https://github.com/pybamm-team/PyBaMM/discussions/2722#discussioncomment-5085461_

### PR DESCRIPTION
**Pull request by jeromtom**
# Description
Changed the >>> prompt to a text prompt and provided instructions on how to run the command in the terminal, to prevent mistakenly running the docstring in the Python interpreter.

Fixes #2728

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.
- [X] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [X] All tests pass: `$ python run-tests.py --all`
- [X] The documentation builds: `$ python run-tests.py --doctest`

### CODE REVIEW

File: pybamm/util.py [hunk 1]
```diff
@@ -333,7 +333,8 @@ def install_jax(arguments=None):  # pragma: no cover
 
     Command Line Interface:
     -----------------------
-    >>> pybamm_install_jax
+    To install compatible versions of jax and jaxlib, run:
```
Comment (Saransh-cpp) [inline pybamm/util.py:336]:
```suggestion
```

**Review by Saransh-cpp — COMMENTED** [review]
Thanks, @jeromtom! Could you also add the function to API docs?


https://pybamm.readthedocs.io/en/latest/source/api/util.html misses the API documentation for install_jax. The function can be added here: https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/api/util.rst

File: pybamm/util.py (hunk 1 above)
Comment (jeromtom) [reply pybamm/util.py]: Hey @Saransh-cpp, good to see you again.
154743c will update the API Docs

File: docs/source/api/util.rst [hunk 2]
```diff
@@ -22,6 +22,8 @@ Utility functions
 
 .. autofunction:: pybamm.get_parameters_filepath
 
+.. autofunction:: pybamm_install_jax
```
Comment (Saransh-cpp) [inline docs/source/api/util.rst:25]: This should be:
```suggestion
.. autofunction:: pybamm.install_jax
```

Comment (jeromtom) [reply docs/source/api/util.rst:25]: What is the suggested change?
image
It seems to be the same.

Comment (Saransh-cpp) [reply docs/source/api/util.rst:25]: `pybamm_install_jax` and `pybamm [dot] install_jax`

**Comment by jeromtom** [comment]
@Saransh-cpp  @tinosulzer. This PR is ready for review and can be merged.

**Comment by Saransh-cpp** [comment]
Thanks, @jeromtom! A minor change -

The docstring of `install_jax` renders weirdly on the website. You might need to check how other docstrings are written and format this one accordingly. I think you can get rid of -
```
Command Line Interface:
-----------------------
```

The rendered docs - https://pybamm--2729.org.readthedocs.build/en/2729/source/api/util.html

image

**Comment by jeromtom** [comment]
Command Line Interface is rendered as a sub-heading. https://docutils.sourceforge.io/docs/user/rst/quickref.html
image
Removing the underscore string would fix it. Doing it now.

**Comment by jeromtom** [comment]
@Saransh-cpp, looks better now rt?

image

File: pybamm/util.py [hunk 3]
```diff
@@ -331,9 +331,9 @@ def install_jax(arguments=None):  # pragma: no cover
...
     Install compatible versions of jax, jaxlib.
 
-    Command Line Interface:
-    -----------------------
-    >>> pybamm_install_jax
+    Command Line Interface::
+
+        $ pybamm_install_jax
 
     optional arguments:
```
Comment (Saransh-cpp) [inline pybamm/util.py:338]: Looks great! Nitpicking a bit -

image

Can the optional arguments render in separate lines?

Comment (jeromtom) [reply pybamm/util.py:338]: Yup @Saransh-cpp, the whole point of docs is to have easy readability so nitpicking will do us good. Will make the commits soon.

**Comment by Saransh-cpp** [comment]
@jeromtom I hope you are building the docs locally to visualize your changes -

```
cd docs
make clean
make html
```

instead of relying on the RTD action.

Also, some general open-source tips -

- While it is okay to have a messed up git commit history, your commit messages should be meaningful (and not `Update file.name`)
- PR titles should ideally describe the changes done in the PR (something like "Fix the API docs of `install_jax`")

**Comment by jeromtom** [comment]
I didn't do it yet. Will do it soon. Thanks for the tips.

**Comment by jeromtom** [comment]
image
image
How do I fix this?

**Comment by Saransh-cpp** [comment]
Have you installed pybamm from the source?

**Comment by jeromtom** [comment]
Yes

**Comment by jeromtom** [comment]
Hey, I have fixed the error, working on the PR now.

**Comment by jeromtom** [comment]
I have fixed the issue but will have to create a new PR.
image

**Comment by jeromtom** [comment]
Jnkjnfjwnekjf

**Comment by jeromtom** [comment]
image

File: pybamm/util.py [hunk 4]
```diff
@@ -331,13 +331,11 @@ def install_jax(arguments=None):  # pragma: no cover
...
-    Command Line Interface:
-    -----------------------
-    >>> pybamm_install_jax
-
-    optional arguments:
-    -h, --help   show help message
-    -f, --force  force install compatible versions of jax and jaxlib
+    Command Line Interface::
+        $ pybamm_install_jax
+    |   optional arguments:
```
Comment (Saransh-cpp) [inline pybamm/util.py:338]: Will render it properly -
```suggestion
    Command Line Interface::

        $ pybamm_install_jax

    |   optional arguments:
```

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @jeromtom! Looks good to me!

**Comment by jeromtom** [comment]
@Saransh-cpp, thank you for the approval. There is one codecov failing check, noticed that some of the earlier merged PRs had this test failing. Is there anything I can do to fix it?

**Comment by tinosulzer** [comment]
In this case it's not related to your changes, sometimes it gets it wrong when another PR is merged in between

**Pull request merged**